t/op/signatures.t

```diff
@@ -1608,6 +1613,82 @@ while(<$kh>) {
             'f($1)';
 }
 
+# check that various uses of @_ inside signatured subs causes "experimental"
+# warnings at compiletime
+{
+    sub warnings_from {
+        my ($code, $run) = @_;
+        my $warnings = "";
+        local $SIG{__WARN__} = sub { $warnings .= join "", @_ };
+        my $cv = eval qq{ sub(\$x) { $code }} or die "Cannot eval() - $@";
+        $run and $cv->(123);
+        return $warnings;
+    }
+
+    sub snailwarns_ok {
```

Comment:
I like these names

Reply:
I do not like any reference to snails for `@_`.  I can remember someone using that terminology at YAPC many years ago -- when I had already been doing Perl for many years -- and thinking, _"What is that guy talking about?"_

When I saw this in my email today, that was still my first reaction -- 22 years on in!  It's too "Perl insider talk" -- for me, and certainly for newcomers.  It's too self-consciously clever and too cute by half.  I object.

Reply:
I don't feel this matters. The only user facing component is the warning, which the user will not care about unless they've been told to disable it, at which point they will know the name.

Reply:
@jkeenan As to the internal function names in the test file - I don't see that it matters. We have tonnes of "sub foo", "sub wibble" and all sorts else worse in .t files. Internal names within files aren't that interesting.

On the specific note of the warning name - it doesn't need to be discoverable, as the name of the warning category is never printed anywhere. The only place a user will find it is in the docs about the specific situation, where they can be told how to turn it off.

But in case there are better ideas there - do you have any better suggestions? My other potential thought was

```perl
no warnings 'experimental::argsarray_in_signatures';
```

but that's even longer a name.

Reply:
alternat ideas: `defav_in_signatures` or `poparray_in_signatures` ?

Reply:
`defav` has a definite meaning to about the 6 people sufficiently familiar with perl core internals, and is just as meaningless to everyone else as `snail` is. Though 4 of those 6 people would point out we don't have a `PL_defav`, what we have is `GvAV(PL_defgv)` and I don't think that will encode as nicely in a warning category ;)

`poparray` I'm not even sure what it's about

Reply:
was trying to see the name of the C variable we used for `GvAV(PL_defgv)`
looks like we have a vote going on with a better choice, thanks